framework/wazuh/core/agent.py

```diff
@@ -190,6 +190,11 @@ def __init__(self, offset=0, limit=common.DATABASE_LIMIT, sort=None, search=None
     def _add_select_to_query(self):
         pass
 
+    def _add_sort_to_query(self):
+        # Consider the option of sort by count
+        self.fields['count'] = 'count(id_group)'
+        super()._add_sort_to_query()
+
```

Comment:
Line 195 is always executed and it updates the `self.fields` dictionary with `{'count': 'count(id_group)'}`. 

Overloading this method would be useful if we had more stuff to do. In this case, including this key and value in the constructor of the `WazuhDBQueryGroup` is a cleaner approach:

https://github.com/wazuh/wazuh/blob/2d8e46a8824274d31db2d49673aa81c282aeba26/framework/wazuh/core/agent.py#L173-L188

`fields={'name': 'name', 'count': 'count(id_group)'}`

Reply:
Update unit tests after the requested change.

Reply:
`self.fields` also affects to `_add_search_to_query()` so is a better option not to include it in the constructor of the `WazuhDBQueryGroup` class

https://github.com/wazuh/wazuh/blob/2d8e46a8824274d31db2d49673aa81c282aeba26/framework/wazuh/core/utils.py#L1268-L1275

Unit tests were reviewed and a new one was included